indra/llimage/llimage.h

```diff
@@ -356,7 +356,9 @@ class LLImageFormatted : public LLImageBase
 	S8 mDecoded;  // unused, but changing LLImage layout requires recompiling static Mac/Linux libs. 2009-01-30 JC
 	S8 mDiscardLevel;	// Current resolution level worked on. 0 = full res, 1 = half res, 2 = quarter res, etc...
 	S8 mLevels;			// Number of resolution levels in that image. Min is 1. 0 means unknown.
-	
+
+	std::mutex mMutex;
```

Comment:
Good, but LLMutex might be better since we've already have the wrapper.

Reply:
Agree, please use LLMutex, and probably a good idea to switch LL_ERRS() to LL_WARNS(), since if section is mutex locked,  mDecoding beeing set is a sign of decoding thread terminating in a wrong state, not running.

Reply:
LLMutex is an extension of an std mutex, unless specific advantages are needed, I'm not sure if there is a point.

Reply:
I believe this ERRS should never happen after this change

And if it happens for some reason, this WARNS could lead to a crash in other thread,
and we won't see the root cause again (WARNS doesn't show the callstack)

ERRS is actually what we need in this case